Riku/src/Frontend/GUIUpdate.h

```diff
@@ -1,25 +1,135 @@
+            if (icon->getProperty("Image").empty())
+            {
+                icon->destroy();
+                resourceElem = my_gui->createWidget("WindowsLook/Static", glm::vec4(x, 0.05f, 0.25f, 0.8f), glm::vec4(0.0f), res.first);
+                icon = my_gui->createWidget("WindowsLook/Label", glm::vec4(0.05f, 0.1f, 0.6f, 0.8f), glm::vec4(0.0f), res.first + "/label");
+                val = my_gui->createWidget("WindowsLook/Label", glm::vec4(0.65f, 0.1f, 0.35f, 0.8f), glm::vec4(0.0f), res.first + "/value");
+                icon->setText(res.first+": ");
+                x += 0.3;
+            }
+            else
+            {
+                icon->setTooltip(0);
+                icon->setTooltipText(res.first); // niestety nie dziala
+                resourceElem = my_gui->createWidget("WindowsLook/Static", glm::vec4(x, 0.05f, 0.15f, 0.8f), glm::vec4(0.0f), res.first);
+                val = my_gui->createWidget("WindowsLook/Label", glm::vec4(0.55f, 0.1f, 0.4f, 0.8f), glm::vec4(0.0f), res.first + "/value");
+                x += 0.2;
+            }             
+            val->setText(std::to_string(res.second));
+            resourceElem->addChild(icon);
+            resourceElem->addChild(val); 
+            resourcesList->addChild(resourceElem);     
+        }
+    }
+    static void LoadIcons(FrontendState& state)
+    {
+        auto resources = state.getResources();
+        for (auto res : resources)
+        {
+            try
+            {
+                CEGUI::GUI::loadIcon(res.first, res.first + ".png"); //TODO wczytywanie r�nych format�w
+                printf("Successfully loaded icon for: %s\n", res.first.c_str());
+            }
+            catch (...) {}            
+        }
+        auto units = state.getUnits();
+        for (auto u : units)
+        {
+            try
+            {
+                CEGUI::GUI::loadIcon(u.get()->getName(), u.get()->getName() + ".png"); //TODO wczytywanie r�nych format�w
+                printf("Successfully loaded icon for: %s\n", u.get()->getName().c_str());
+            }
+            catch (...) {}
+        }
+    }
+    static void CreateUnits(CEGUI::GUI* my_gui, const CEGUI::String& unitsListName, FrontendState& state, int& focusedUnitIndex)
+    {
+        auto unitsList = static_cast<CEGUI::ScrollablePane*>(my_gui->getWidgetByName(unitsListName));
+        auto player_units = state.getUnits();
+        float y = 0.05f;
+        std::map<std::string, int> repeats;
+        std::vector<std::string> unitNames;
+        for (auto u : player_units)
+        {
+            std::string name = u.get()->getName();
+            if (repeats.find(name) == repeats.end())
+                repeats.insert(std::pair<std::string, int>(name, 0));
+            repeats[name]++;
+            int count = repeats[name];
+            name = name + std::to_string(count);
+            unitNames.push_back(name);
+        }
+        int i = 0;
+        for(auto u : player_units)
+        {
```

Comment:
Dlaczego to są dwa fory zamiast jednego?

Reply:
A nie można tego zobić smart_ptr'em?

Reply:
wyjaśnisz mi na spotkaniu smart pointery, bo nie umiem zrobić żeby działało? ;p